src/uct/sm/mm/base/mm_ep.c

```diff
@@ -64,130 +120,55 @@ static void uct_mm_ep_signal_remote(uct_mm_ep_t *ep)
 {
-    uct_mm_iface_t *iface           = ucs_derived_of(params->iface, uct_mm_iface_t);
-    const uct_mm_iface_addr_t *addr = (const void *)params->iface_addr;
-
+    uct_mm_iface_t            *iface = ucs_derived_of(params->iface, uct_mm_iface_t);
+    const uct_mm_iface_addr_t *addr  = (const void *)params->iface_addr;
     ucs_status_t status;
-    size_t size_to_attach;
+    void *fifo_ptr;
 
     UCT_EP_PARAMS_CHECK_DEV_IFACE_ADDRS(params);
     UCS_CLASS_CALL_SUPER_INIT(uct_base_ep_t, &iface->super.super);
 
-    /* Connect to the remote address (remote FIFO) */
-    /* Attach the address's memory */
-    size_to_attach = UCT_MM_GET_FIFO_SIZE(iface);
-    status         = uct_mm_iface_mapper_call(iface, attach, addr->id,
-                                              size_to_attach,
-                                              (void *)addr->vaddr,
-                                              &self->mapped_desc.address,
-                                              &self->mapped_desc.cookie,
-                                              iface->path);
+    kh_init_inplace(uct_mm_remote_seg, &self->remote_segs);
+    ucs_arbiter_group_init(&self->arb_group);
+
+    /* Attach the remote FIFO, use the same method as bcopy descriptors */
+    status = uct_mm_ep_get_remote_seg(self, addr->fifo_seg_id, (void*)addr->vaddr,
+                                      UCT_MM_GET_FIFO_SIZE(iface), &fifo_ptr);
     if (status != UCS_OK) {
-        ucs_error("failed to connect to remote peer with mm. remote mm_id: %zu",
-                   addr->id);
-        return status;
+        goto err_destroy_remote_segs;
     }
 
-    self->mapped_desc.length     = size_to_attach;
-    self->mapped_desc.mmid       = addr->id;
-
-    /* point the ep->fifo_ctl to the remote fifo.
-      * it's an aligned pointer to the beginning of the ctl struct in the remote FIFO */
-    self->fifo_ctl        = uct_mm_set_fifo_ctl(self->mapped_desc.address);
+    /* Initialize remote FIFO control structure */
+    uct_mm_iface_set_fifo_ptrs(fifo_ptr, &self->fifo_ctl, &self->fifo_elems);
     self->cached_tail     = self->fifo_ctl->tail;
     self->signal.addrlen  = self->fifo_ctl->signal_addrlen;
     self->signal.sockaddr = self->fifo_ctl->signal_sockaddr;
 
-    /* Make sure the fifo ctrl is aligned */
-    ucs_assert_always(((uintptr_t)self->fifo_ctl % UCS_SYS_CACHE_LINE_SIZE) == 0);
-
-    /* set the ep->fifo ptr to point to the beginning of the fifo elements at
-     * the remote peer */
-    uct_mm_set_fifo_elems_ptr(self->mapped_desc.address, &self->fifo);
-
-    /* Initiate the hash which will keep the base_adresses of remote memory
-     * chunks that hold the descriptors for bcopy. */
-    sglib_hashed_uct_mm_remote_seg_t_init(self->remote_segments_hash);
-
-    ucs_arbiter_group_init(&self->arb_group);
-
-    ucs_debug("mm: ep connected: %p, to remote_shmid: %zu", self, addr->id);
-
     return UCS_OK;
+
```

Comment:
can you please add a debug message about the creation of the ep and its details?